test/common/tmpdir.js

```diff
@@ -51,8 +73,11 @@ function onexit() {
       console.error('See http://nfs.sourceforge.net/#faq_d2 for details.');
     }
 
-    console.error();
-    throw e;
+    // Additionally logging err, just in case throwing err gets overshadowed by
+    // an error from a test failure.
+    console.error(err);
+
+    throw err;
```

Comment:
This is supposed to prevent `err` from being logged twice when the test succeeds but the `rmSync` call of `tmpdir` fails. Is there a better way of doing this?
```suggestion
    // Manually logging err instead of throwing it, so that it doesn't get
    // overshadowed by an error from a test failure.
    console.error(err);

    if (!process.exitCode) {
      process.exitCode = 1;
    }
```

Reply:
You can use `||=` to avoid the `if` statement. If think you also need to add a comment explaining why this is necessary.
```suggestion
    // Additionally logging err, just in case throwing err gets overshadowed by
    // an error from a test failure.
    console.error(err);

    // Setting process exit code instead of throwing the error so `tmpdir` can be deleted or something like that.
    process.exitCode ||= 1;
```

Reply:
Wow, `||=` must be really new.  In latest LTS `14.17.0` it's a SyntaxError 🤔
It will have to be manually backported (if at all).

Reply:
Using `||=` here causes `build-docs` to fail:
```console
/home/runner/work/node/node/test/common/tmpdir.js:82
    process.exitCode ||= 1;
                     ^^^

SyntaxError: Unexpected token '||='
```
Added the `if` equivalent back in.